iothub/service/src/Messaging/MessagesClient.cs

```diff
@@ -391,17 +392,16 @@ private void OnConnectionClosed(object sender, EventArgs e)
         {
             if (((AmqpObject)sender).TerminalException is AmqpException exception)
             {
-                ErrorContext errorContext = AmqpClientHelper.GetErrorContextFromException(exception);
-                ErrorProcessor?.Invoke(errorContext);
-                Exception exceptionToLog = errorContext.IotHubServiceException;
+                IotHubServiceException mappedException = AmqpClientHelper.GetIotHubExceptionFromAmqpException(exception);
+                ErrorProcessor?.Invoke(new MessagesClientError(mappedException));
```

Comment:
Shouldn't this be awaited?

Reply:
This callback method is sync since the AMQP library demands it. We can either fire-and-forget here or we can block on this call to make it sync. Thoughts?

Reply:
It's worth noting that this callback doesn't return anything, so we don't _need_ to wait on it.

Reply:
If that processor throws? Then what?

Reply:
If we block and wait for this invocation, we'd need to wrap it in a try/catch so it doesn't crash the SDK. If we don't wait for it, then it doesn't matter whether their callback code throws or not.

Reply:
I'm inclined to not wait for the callback here since it risks deadlock

Reply:
k